### ISSUE TITLE
error building site didn't fail Travis

### ISSUE THREAD

**Issue by SethTisue** [scala/scala-lang#520]
see https://github.com/scala/scala-lang/pull/516 for details

why didn't Jekyll give a nonzero exit code here?

~~also, 748c6353bd0217ccf8710d38d13f3f2f5eab72e0 was reverted, should be redone~~ (#525)

**Comment by jarrodwb** [comment]
Any word on Travis? I find it really strange that it did not catch it.

**Comment by SethTisue** [comment]
I haven't had time to investigate.

**Comment by SethTisue** [comment]
problem is reproducible locally with:

```
% bundle exec jekyll build && echo success
Configuration file: /Users/tisue/scala-lang/_config.yml
            Source: /Users/tisue/scala-lang
       Destination: /Users/tisue/scala-lang/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
Error reading file /Users/tisue/scala-lang/download/index.md: (<unknown>): did not find expected ',' or ']' while parsing a flow sequence at line 7 column 17 
                    done in 7.807 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
success
```

**Comment by SethTisue** [comment]
I had thought adding this to `_config.yml` might help:

```
liquid:
  error_mode: strict
```

but it didn't.

**Comment by ashawley** [comment]
I ran across jekyll/jekyll#5257

Apparently,

>  We only throw on fatal errors and the inability to process a file is not fatal to us so we processed what we could and gave you the error so that you could fix it.

Scrolling further, it seems it might be a flaw in Jekyll and Ruby and not really a design decision:

> We made the decision here some years ago (maybe even before I joined?) to warn here. We have never failed loudly here. I shudder to think how many tickets we would receive because "all of a sudden my site didn't build OH NOES!" In certain cases, failing gracefully is the right thing to do. [...]

> I would be up for raising an error here. Unfortunately we don't have a mechanism for raising an error but continuing the process of building the rest of the site and still returning a 1 exit code. If we fail at all, we'll have to fail mid-build.

**Comment by SethTisue** [comment]
perhaps we will have to capture and inspect Jekyll's output

**Comment by charafau** [comment]
I havent used it personally but on [jekyll's docs](https://jekyllrb.com/docs/continuous-integration/) mention `html-proofer` for checks. Maybe we can give it a try ?

**Comment by SethTisue** [comment]
re: html-proofer, that's #535 now

**Comment by SethTisue** [comment]
fixed by #537
